### ISSUE TITLE
"The Site Kit authentication service"

### ISSUE THREAD

**Issue by eclarke1** [google/site-kit-wp#3449]
## Feature Description

- Should explain the role of “Site Kit Service” (a.k.a. "The Proxy") which facilitates centralized OAuth in an otherwise decentralized system
- Should also explain other supplementary roles it plays (e.g. remote control of feature flags, remote triggering of notifications etc.)
- Should touch at a high-level on how it internally works to indicate e.g. what it has access to (e.g. tokens are not stored, so user requests only work with the user involved; Site Kit’s access tokens can be used against Google APIs directly while Site Kit’s refresh tokens only work against the service since they are encrypted)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by eclarke1** [comment]
@felixarntz @aaemnnosttv @marrrmarrr this was one of the 4 articles voted highest priority by engineers so if possible to progress this through the workflow along with the others labelled 'Type: Docs' and have P1 assigned.

**Comment by aaemnnosttv** [comment]
@felixarntz This looks good to me overall. Should we include a section about its limitations/requirements for using it? Perhaps this might mostly link to "Setting up a local development environment" but it also relates to use on a [staging site](https://sitekit.withgoogle.com/documentation/using-site-kit-on-a-staging-environment/).

**Comment by felixarntz** [comment]
@aaemnnosttv Great idea! I added an extra sub-heading level item that goes into more depth for the exact requirements and in that also hints at what this means for local setup. It will mostly be a brief mention for the latter though and then point to the (future) article about the local development setup.

**Comment by aaemnnosttv** [comment]
IB ✅ 👍

**Comment by eclarke1** [comment]
@aaemnnosttv please could we add an estimate to this one? I understand it's writing documentation so let's consider a round of review and iteration also. cc @fhollis

**Comment by aaemnnosttv** [comment]
@eclarke1 Added!

@felixarntz let me know what you think. I don't think it would be more than 11 but that also sounds like a lot for a document. Then again, we haven't done any of these yet and this is one of the more complex articles to write.

**Comment by felixarntz** [comment]
@aaemnnosttv 7 sounds good to me generally, I'd imagine 3-4 hours for the actual writing and then some buffer for review and iterations. Since this is one of the very first, I'm going to put 11 just to be safe. The first posts may need to be scrutinized more than later ones, to set a good example for the later ones.

@eclarke1 I actually intend to work on this one myself, so maybe we can already get one more of the others into next sprint? I'd say just one more though, since then those first two as mentioned above could serve as examples later, so I wouldn't want to have too many "first articles" at the same time, to avoid them all going in different directions.

**Comment by eclarke1** [comment]
Certainly, the others are in IB so I will loop in @fhollis here to see if we can get another pushed through for Sprint 60. Thanks.
